include/aspect/particle/property/interface.h

```diff
@@ -178,6 +178,19 @@ namespace aspect
           need_update () const;
 
           /**
+           * Return, which data has to be provided to update the property.
+           * Note that particle properties can only ask for no data, the
+           * solution values, and the solution gradients and all other update
+           * flags will throw exceptions.
+           *
+           * @return A vector with as many entries as solution components.
+           * Each entry contains the necessary update flags for this component.
```

Comment:
Ah, I see.

But do you actually need this? I mean, it is *computing* the values or gradients that is expensive, not necessarily passing them around. So wouldn't it be enough if every class just provides the *sum* of all flags?